File: src/main/java/arcana/common/research/sections/RequirementSection.java
package arcana.common.research.sections;

import arcana.common.research.BookSection;
import arcana.common.research.Requirement;
import com.google.gson.JsonObject;

import java.util.List;

public class RequirementSection extends BookSection {
    List<Requirement> requirements;

    @Override
    public void deserialize(JsonObject json) {
        requirements = Requirement.parse(json);
    }
}

File: src/main/kotlin/arcana/common/research/sections/RequirementSection.kt
package arcana.common.research.sections

import arcana.common.research.BookSection
import arcana.common.research.Requirement
import com.google.gson.JsonObject

class RequirementSection : BookSection() {
    lateinit var requirements: List<Requirement>
    override fun deserialize(json: JsonObject) {
        requirements = Requirement.parse(json)
    }
}